### PR TITLE
:bug: Fix release-notes in Makefile

### PR DESCRIPTION
**Pull request by peppi-lotta**
**What this PR does / why we need it**: Fix the path release binary is created in. Add more logging and documentation to reduce user error when using release-notes action from Makefile.

### CODE REVIEW

**Comment by tuminoid** [comment]
I'll take a look tomorrow

File: docs/releasing.md [hunk 1]
```diff
@@ -46,6 +46,9 @@ by using following command `git remote -v`.
 - Create a new branch for the release notes**:
   `git checkout -b release-notes-1.x.x origin/main`
 
+- Fetch upstream (`metal3-io`): `git fetch upstream`
```
Comment (tuminoid) [inline docs/releasing.md:49]: Put this on L41 where it talks about remotes.

File: hack/tools/release/notes.go [hunk 2]
```diff
@@ -125,16 +125,16 @@ func isMinor(tag string) bool {
...
 	lastTag, err := lastTag(latestTag)
 	if err != nil {
-		log.Fatal("Failed to get lastTag \n")
+		log.Fatalf("Failed to get lastTag: %v", err)
 	}
 
 	commitHash, err := getCommitHashFromNewTag(latestTag)
 	if err != nil {
-		log.Fatalf("Failed to get commit has from latestTag %s", latestTag)
+		log.Fatalf("Failed to get commit has from latestTag %s: %v", latestTag, err)
```
Comment (mquhuy) [inline hack/tools/release/notes.go:137]: "hash", not "has"

**Comment by tuminoid** [comment]
/override metal3-centos-e2e-integration-test-main metal3-ubuntu-e2e-integration-test-main

**Review by tuminoid — COMMENTED** [review]
/approve

Manually tested, it works as expected.

**Pull request merged**